unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarShape/AvatarShape.cs

```diff
@@ -0,0 +1,426 @@
+            entity.OnTransformChange += OnEntityTransformChanged;
+
+            var wearableItems = model.Wearables.ToList();
+            wearableItems.Add(model.BodyShape);
+
+            //temporarily hardcoding the embedded emotes until the user profile provides the equipped ones
+            var embeddedEmotesSo = Resources.Load<EmbeddedEmotesSO>("EmbeddedEmotes");
+            wearableItems.AddRange(embeddedEmotesSo.emotes.Select(x => x.id));
+
+            if (avatar.status != IAvatar.Status.Loaded || needsLoading)
+            {
+                loadingCts?.Cancel();
+                loadingCts?.Dispose();
+                loadingCts = new CancellationTokenSource();
+                
+                await LoadAvatar(wearableItems);
+            }
+            
+            avatar.PlayEmote(model.ExpressionTriggerId, model.ExpressionTriggerTimestamp);
+
+            UpdatePlayerStatus(entity,model);
+
+            onPointerDown.Initialize(
+                new OnPointerDown.Model()
+                {
+                    type = OnPointerDown.NAME,
+                    button = WebInterface.ACTION_BUTTON.POINTER.ToString(),
+                    hoverText = "view profile"
+                },
+                entity, player
+            );
+
+            avatarCollider.gameObject.SetActive(true);
+
+            EnablePassport();
+
+            onPointerDown.SetColliderEnabled(isGlobalSceneAvatar);
+            onPointerDown.SetOnClickReportEnabled(isGlobalSceneAvatar);
+        }
+
+        private async UniTask LoadAvatar(List<string> wearableItems)
+        {
+            try
+            {
+                //TODO Add Collider to the AvatarSystem
+                //TODO Without this the collider could get triggered disabling the avatar container,
+                // this would stop the loading process due to the underlying coroutines of the AssetLoader not starting
+                avatarCollider.gameObject.SetActive(false);
+
+                SetImpostor(model.Id);
+                playerName.SetName(model.Name);
+                playerName.Show(true);
+                await avatar.Load(wearableItems, new AvatarSettings
+                {
+                    playerName = model.Name,
+                    bodyshapeId = model.BodyShape,
+                    eyesColor = ProtoConvertUtils.PBColorToUnityColor(model.EyeColor),
+                    skinColor = ProtoConvertUtils.PBColorToUnityColor(model.SkinColor),
+                    hairColor = ProtoConvertUtils.PBColorToUnityColor(model.HairColor),
+                }, loadingCts.Token);
+            }
+            catch (OperationCanceledException)
+            {
+                Cleanup();
+                throw;
+            }
+            catch (Exception e)
+            {
+                Cleanup();
+                Debug.Log($"Avatar.Load failed with wearables:[{string.Join(",", wearableItems)}] for bodyshape:{model.BodyShape} and player {model.Name}");
+            }
```

Comment:
You have to throw the exception otherwise we won't know the problem. 

Do it like this: 
```
if (e.InnerException != null)
    ExceptionDispatchInfo.Capture(e.InnerException).Throw();
else
    throw;
```